ProjectCoimbra.UWP/Project.Coimbra.Midi/MidiEngine.cs

```diff
@@ -2,353 +2,395 @@
-                {
-                    if (!this.Instruments[currentEvent.Channel].Contains(programNumber))
-                    {
-                        this.Instruments[currentEvent.Channel].Add(programNumber);
-                    }
-                }
-                else
-                {
-                    this.Instruments[currentEvent.Channel] = new List<SevenBitNumber> { programNumber };
-                }
-            }
-        }
-
-        /// <summary>
-        /// Invoked whenever the playback is stopped.
-        /// </summary>
-        private void OnStop()
-        {
-            if (this.isStopped)
-            {
-                this.playback?.Start();
-            }
-            else
-            {
-                this.playback?.Stop();
-            }
-
-            this.isStopped = !this.isStopped;
-        }
-
-        private void Media_PausePressed(object sender, EventArgs e) =>
-            this.OnStop();
-
-        private void Media_PlayPressed(object sender, EventArgs e)
-        {
-            if (this.isStopped)
-            {
-                this.OnStop();
-            }
-        }
-
-        private void Playback_NotesDisplayFinished(object sender, MarkablePlaybackEventArgs e)
-        {
-            var offTest = ((NoteOffEvent)e.PlaybackEvent.Event).NoteNumber;
-            this.notesOnDisplay[offTest] = new ConcurrentQueue<MarkablePlaybackEvent>();
-        }
-
-        private void Playback_NotesDisplayStarted(object sender, MarkablePlaybackEventArgs e) =>
-            this.notesOnDisplay[((NoteOnEvent)e.PlaybackEvent.Event).NoteNumber].Enqueue(e.PlaybackEvent);
-
-        private void RenderCurrentNotes()
-        {
-            while (!this.isStopped)
-            {
-                var task = this.RenderCurrentNotesAsyncEvent?.Invoke(this.notesOnDisplay);
-                task?.ConfigureAwait(true).GetAwaiter().GetResult();
-                Thread.Sleep(TimeSpan.FromMilliseconds(200));
-            }
-        }
-
-        private void Playback_SelectedTrackNotification(object sender, TrackNotificationEventArgs e) =>
-            this.PlaybackSelectedTrackNotificationEvent?.Invoke(e);
-
-        private void Playback_Finished(object sender, EventArgs e)
-        {
-            // The finished event fires 5 seconds before the last note finished playing,
-            // due to difference between display and audio, leaving one second of buffer.
-            Task.Delay(6000).ContinueWith(t =>
-            {
-                this.startTimer?.Dispose();
-                this.threadCancellationToken.Cancel();
-                this.startThreadCancellationToken.Cancel();
-                this.playback?.Dispose();
-                this.outputDevice?.Dispose();
-
-                PlaybackFinished?.Invoke(this, null);
-            });
-        }
-
-        private void OutputDevice_EventSent(object sender, MidiEventSentEventArgs e) =>
-            this.OutputDeviceEventSentEvent?.Invoke(e);
-    }
+	using System;
```

Comment:
It looks like you've changed all of the line endings in this file. Can you ensure they are as they were before, to avoid unnecessary repo churn?

Reply:
Or I think you might have changed all the spaces to tabs

Reply:
Sorry VS messed that up. Fixing for all files now.